plugins/mux/serve_interceptor.go

```diff
@@ -58,8 +57,19 @@ func (n *ServeHTTPInterceptor) AfterInvoke(invocation operator.Invocation, resul
 	return nil
 }
 
+func newWriterWrapper(rw interface{}) *writerWrapper {
+	writer := rw.(http.ResponseWriter)
+	hijacker := rw.(http.Hijacker)
```

Comment:
Is any possible the agent will got panic when cast failure? Please do more explain on this.

Reply:
> Is any possible the agent will got panic when cast failure? Please do more explain on this.

If it were the original `http.ResponseWirter` then the panic would definitely not happen. 

https://github.com/golang/go/blob/master/src/net/http/server.go#L2155

Reply:
Then, please added a cast check please, some third-party library may change this.

Reply:
sure.